messagelogger.go

```diff
@@ -32,7 +33,27 @@ func messageDelete(s *discordgo.Session, m *discordgo.MessageDelete) {
 			log.Println("Recovered from panic in messageDelete", r)
 		}
 	}()
-
+	var deleter string
+	// get audit log info
+	auditLog, err := s.GuildAuditLog(m.GuildID, "", "", discordgo.AuditLogActionMessageDelete, 1)
+	if err != nil {
+		log.Printf("Failed to check audit log: %s", err.Error())
+		return
+	}
+	// get audit log entries
+	for _, v := range auditLog.AuditLogEntries {
```

Comment:
As you are only requesting a single Entry from the log, using a loop here is somewhat confusing.
I'd propose something like this:
```go
	if len(auditLog.AuditLogEntries) == 0 {
		log.Printf("Received no audit-log entries")
	}
	lastEntry := auditLog.AuditLogEntries[0]
	if lastAuditIds[m.GuildID] == lastEntry.ID {
		deleter = m.Author.Username + "#" + m.Author.Discriminator
	} else {
		lastAuditIds[m.GuildID] = lastEntry.ID

		for _, u := range auditLog.Users {
			if u.ID == lastEntry.UserID {
				deleter = u.Username + "#" + u.Discriminator
			}
		}
	}

```